docs/roleassumption.md

```diff
@@ -0,0 +1,393 @@
+# Creating clusters using cross account role assumption using kiam
+
+This document outlines the list of steps to create the target cluster via cross account role assumption using KIAM. 
+KIAM lets the controller pod(s) to assume an AWS role that enables them create AWS resources necessary to create an
+operational cluster. This way we wouldn't have to mount any AWS credentials or load environment variables to 
+supply AWS credentials to the CAPA controller. This is automatically taken care by the KIAM components.
+Note: If you dont want to use KIAM and rather want to mount the credentials as secrets, you may still achieve cross 
+account role assumption by using multiple profiles. 
+
+### Glossory
+
+* Target Account - The account where the cluster is created
+* Source Account - The AWS account where the CAPA controller runs. 
+
+## Assumptions
+1. The CAPA controllers are running in 1 AWS account and you want to create the target cluster in another AWS account.
+(We could also use this doc to create a cluster in the same account)
+2. This assumes that you start with no existing clusters. 
+
+## High level steps
+
+1. Creating a bootstrap/management cluster in AWS - This can be done by running the phases in clusterctl
+    * Uses the existing provider components yaml
+2. Setting up cross account roles
+3. Deploying the Kiam server/agent
+4. Create the target cluster (through kiam)
+    * Uses different provider components with no secrets and annotation to indicate the IAM Role to assume. 
+
+## 1. Creating the bootstrap cluster in AWS
+Using clusterctl command we can create a new cluster in AWS which in turn will act as the 
+bootstrap cluster to create the target cluster(in a different AWS account. This can be achieved by using the phases in 
+clusterctl to perform all the steps except the pivoting. This will provide us with a bare-bones functioning cluster that 
+we can use as a bootstrap cluster. 
+To begin with follow the steps in this getting started guide
+(https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/getting-started.md) to setup the environment
+except for creating the actual cluster. Instead follow the steps below to create the cluster.
+
+create a new cluster using kind for bootstrapping purpose by running:
+```$xslt
```

Comment:
I don't know what the `$xslt`. Can you please tell me what that is?

Reply:
It should have been $sh. $xslt was added as the default code for mark up. Its a XML format and used to convert one xml document into another.